### ISSUE TITLE
Use any Auth Method via annotation

### ISSUE THREAD

**Issue by pbar1** [hashicorp/vault-k8s#215]
**Is your feature request related to a problem? Please describe.**
Out of the box, the Vault Agent Injector does not support other Auth Methods than the `kubernetes` method using annotations alone.

**Describe the solution you'd like**
It would be great to be able to configure the Auth Method of the Agent that gets injected based on annotation alone. For example, such annotations as:
```
vault.hashicorp.com/auth-type: cert             # new annotation
vault.hashicorp.com/auth-path: auth/cert
vault.hashicorp.com/auth-config-name: pki-test  # new annotation
vault.hashicorp.com/ca-cert: /certs/ca.pem
vault.hashicorp.com/client-cert: /certs/client.pem
vault.hashicorp.com/client-key: /certs/client-key.pem
```

Should be able to generate config that looks like this:
```
{
  "auto_auth": {
    "method": {
      "type": "cert",
      "mount_path": "auth/cert",
      "config": {
        "name": "pki-test"
      }
    },
    "sink": <omitted for brevity>
  },
  "exit_after_auth": false,
  "pid_file": "/home/vault/.pid",
  "vault": {
    "address": "https://vault.vault.svc.cluster.local:8200",
    "ca_cert": "/certs/ca.pem",
    "client_cert": "/certs/client.pem",
    "client_key": "/certs/client-key.pem",
  },
  "template": <omitted for brevity>
}
```

**Describe alternatives you've considered**
The obvious alternative is using the `vault.hashicorp.com/agent-configmap` annotation to configure the agent from an existing ConfigMap. However, this solution requires the ConfigMap in question to already exist in order to be consumed - to use it in a preexisting Helm chart for example, one would have to modify the chart to explicitly create the ConfigMap explicitly. This isn't very declarative and violates the decorator-like pattern of annotations if used extensively.

**Additional context**
I've submitted a pull request extending the existing code to support this feature, #213. Let me know how I can help to get this feature merged. More details are in that PR.

**Comment by pbar1** [comment]
#213

**Comment by pbar1** [comment]
Fixed by #213!
